src/Microsoft.FeatureManagement/Targeting/TargetingEvaluator.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using Microsoft.FeatureManagement.FeatureFilters;
+using System;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Microsoft.FeatureManagement.Targeting
+{
+    static class TargetingEvaluator
+    {
+        private static StringComparison ComparisonType(bool ignoreCase) => ignoreCase ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal;
+
+        public static bool IsTargeted(TargetingFilterSettings settings, ITargetingContext targetingContext, bool ignoreCase, string hint)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+
+            if (targetingContext == null)
+            {
+                throw new ArgumentNullException(nameof(targetingContext));
+            }
+
+            if (!TryValidateSettings(settings, out string paramName, out string reason))
+            {
+                throw new ArgumentException(reason, paramName);
+            }
+
+            //
+            // Check if the user is being targeted directly
+            if (targetingContext.UserId != null &&
+                settings.Audience.Users != null &&
+                settings.Audience.Users.Any(user => targetingContext.UserId.Equals(user, ComparisonType(ignoreCase))))
+            {
+                return true;
+            }
+
+            //
+            // Check if the user is in a group that is being targeted
+            if (targetingContext.Groups != null &&
+                settings.Audience.Groups != null)
+            {
+                foreach (string group in targetingContext.Groups)
+                {
+                    GroupRollout groupRollout = settings.Audience.Groups.FirstOrDefault(g => g.Name.Equals(group, ComparisonType(ignoreCase)));
+
+                    if (groupRollout != null)
+                    {
+                        string audienceContextId = $"{targetingContext.UserId}\n{hint}\n{group}";
```

Comment:
Probably not a new issue... if a user chooses to ignore case, we still generate case sensitive id. Should we take this opportunity to fix it?

Reply:
Yes, I think we should.

Reply:
Updated, now user id and group are normalized if case is ignored.